src/sentry/tagstore/legacy/backend.py

```diff
@@ -63,33 +82,18 @@ def mark_deletion_in_progress(self, instance_list):
                     if instance.status != TagKeyStatus.DELETION_IN_PROGRESS:
                         instance.update(status=TagKeyStatus.DELETION_IN_PROGRESS)
 
-        default_manager.register(TagKey, TagKeyDeletionTask)
-        default_manager.add_dependencies(Project, [
+        deletion_manager.register(TagKey, TagKeyDeletionTask)
+        deletion_manager.add_dependencies(Project, [
             lambda instance: ModelRelation(TagKey, {'project_id': instance.id}),
             lambda instance: ModelRelation(TagValue, {'project_id': instance.id}),
             lambda instance: ModelRelation(GroupTagKey, {'project_id': instance.id}),
             lambda instance: ModelRelation(GroupTagValue, {'project_id': instance.id}),
         ])
 
-        self.setup_cleanup(
-            tagvalue_model=TagValue,
-            grouptagvalue_model=GroupTagValue,
-            eventtag_model=EventTag,
-        )
-
-        self.setup_merge(
-            grouptagkey_model=GroupTagKey,
-            grouptagvalue_model=GroupTagValue,
-        )
-
-        self.setup_tasks(
-            tagkey_model=TagKey,
-        )
+        super(LegacyTagStorage, self).setup_deletions(**kwargs)
```

Comment:
If ordering doesn't matter here (I don't think it does, but I could be wrong and didn't spend much time thinking about it), I'd prefer this to be at the top of this method so that it's clearer at a glance that this is extending the behavior of the superclass rather than replacing it. Same goes for `setup_receivers`.